### DISCUSSION TITLE
Issues with navigation to same route with differnent parameter values

### DISCUSSION THREAD

**Discussion by JonWatkins** [vikejs/vike#1783] [Discussion]
Lets say that I am on this page `/admin/comment/1234` but on this page there is a link to `/admin/comment/4356` the page content and pageContext doesnt change?

im trying to navigate using `await navigate("/admin/comment/4356")`

I can see that the new pageContext is fetched. but i cant find a way to get the page to update correctly.  have tried many different ways to in the page compnent to check for changes. We are using the +data hook to fetch the data for the page.

**Comment by JonWatkins** [comment]
I have even tried to bypass vike routing, and use window.location and even tried to force a page reload. but nothing gets the page content to update.

**Comment by brillout** [comment]
I guess you don't use `vike-{react,vue,solid}`? I'm inclined to think it's a user land issue, not a Vike issue. But if you do believe it to be Vike's "fault" then minimal reproduction welcome and I'll have a look at it.
